fast_plotter/postproc/stages.py

```diff
@@ -22,6 +22,11 @@ def __init__(self, **kwargs):
         self.kwargs = kwargs
         self.func = getattr(functions, self.func)
         self.doc = self.func.__doc__
+        if "apply_if" in kwargs:
+            self.apply_if = kwargs['apply_if']
+            kwargs.pop("apply_if", None)
+        else:
+            self.apply_if = False
```

Comment:
Is there a cleaner way of doing this bit? `apply_if` currently is used as an argument in a given stage, so needs to be removed from the kwargs for that stage and applied to `self`

Reply:
You already use `kwargs.pop` with a second argument - that second argument provides the default value in the case where `"apply_if"` isn't encountered in `kwargs` items. That essentially does this entire block in one check and since the code that uses `apply_if` doesn't distinguish between `False` and `None` you could just replace all 5 lines of this edit with:

```python
            self.apply_if = kwargs.pop("apply_if", None)
```

Reply:
Ah, I wasn't sure if `kwargs` would be updated to remove `apply_if` if it was it was assigned to something else